### DISCUSSION TITLE
OnError callback with SolitoImage

### DISCUSSION THREAD

**Discussion by jetsbee** [nandorojo/solito#423] [Q&A]
It would be great if SolitoImage has onError callback.
So fallback image feature can be implemented with SolitoImage.
(Ref. https://stackoverflow.com/a/66953317)

Is it hard for SolitoImage to support onError callback?
And is there any other way to implement fallback feature without onError callback?
I thought getting base64 string from url with networking like fetch or axios, but I'm not sure it's ok.

**Comment by nandorojo** [comment]
does the web image support this

**Reply by jetsbee** [reply]
Docs say both Next.js Image and Expo Image support onError props.
Here's references.
- [Next.js with pages router](https://nextjs.org/docs/pages/api-reference/components/image#onerror)
- [Next.js with app router](https://nextjs.org/docs/app/api-reference/components/image#onerror)
- [Expo image](https://docs.expo.dev/versions/latest/sdk/image/#onerror)

**Reply by nandorojo** [reply]
and it’s not working with solito?

**Reply by jetsbee** [reply]
I think SolitoImage does not support onError props.

ref. https://solito.dev/usage/image#props

**Reply by nandorojo** [reply]
did you create this discussion without trying it first? i assumed you tried it already. is it just missing types?

**Reply by jetsbee** [reply]
I just checked now. It's missing types.
Should I just ignore the message?

Here's message from typescript.
```
Type '{ src: string; alt: string; fill: true; onError: () => void; }' is not assignable to type 'IntrinsicAttributes & (SolitoImageProps & RefAttributes<Image>)'.
  Property 'onError' does not exist on type '(IntrinsicAttributes & Pick<Omit<DetailedHTMLProps<ImgHTMLAttributes<HTMLImageElement>, HTMLImageElement>, "height" | ... 5 more ... | "srcSet"> & { ...; } & RefAttributes<...>, "aria-label" | ... 62 more ... | "unoptimized"> & ... 4 more ... & RefAttributes<...>) | (IntrinsicAttributes & ... 5 more ... & RefAttribu...'.
```
